lib/query-interface.js

```diff
@@ -883,6 +883,25 @@ module.exports = (function() {
     return this.sequelize.query(sql, options);
   };
 
+  QueryInterface.prototype.deferConstraints = function (transaction, options) {
+    if (!transaction || !(transaction instanceof Transaction)) {
```

Comment:
You do not have to check for `!transaction`. `undefined instanceof something` will evaluate to false.